src/opera/parser/tosca/csar.py

```diff
@@ -1,19 +1,92 @@
+import shutil
+import yaml
+
 from pathlib import Path
 from tempfile import TemporaryDirectory
 from zipfile import ZipFile
 
-import yaml
 from opera.error import ParseError
 
 
 class CloudServiceArchive:
-    def __init__(self, csar_name, csar_folder_path):
+    def __init__(self, csar_name, csar_folder_path=None):
         self._csar_name = csar_name
         self._csar_folder_path = csar_folder_path
         self._tosca_meta = None
         self._root_yaml_template = None
         self._metadata = None
 
+    def package_csar(self, output, service_template=None, csar_format="zip"):
+        try:
+            root_yaml_files = []
+            root_yaml_files.extend(Path(self._csar_name).glob('*.yaml'))
+            root_yaml_files.extend(Path(self._csar_name).glob('*.yml'))
+
+            if len(root_yaml_files) != 1:
+                raise ParseError("There should be one root level yaml "
+                                 "file in the root of the CSAR: {}."
+                                 .format(list(map(str, root_yaml_files))),
+                                 self)
```

Comment:
I'd suggest an additional parameter to `package` that lets you choose one of the (more than one) files, falling back to selecting this (only) one if that isn't specified. That would be in line with my expectation of `package` just mainly building the metadata file.

Reply:
Oh, I just noticed that this argument exists as `-t/--service-template`. The code below still fails if there is more than one yaml file at the root though.

Reply:
Hmm yes we have an option (`-t/--service-template`) to point `opera package` command to the selected TOSCA service template. But this will still need some reshaping. According to [TOSCA v1.3](https://docs.oasis-open.org/tosca/TOSCA-Simple-Profile-YAML/v1.3/cos01/TOSCA-Simple-Profile-YAML-v1.3-cos01.html#_Toc26969474) there are two possible CSAR cases:

- CSAR with `TOSCA.meta`:  here you are allowed to have more than one YAML file in the root of the CSAR (because you already have a reference to the entrypoint in the `Entry-Definitions` keyword) 
- CSAR without `TOSCA.meta`: here you should have only one YAML file at the root of the CSAR that contains the `metadata` YAML map entry.

So, with `opera package` we are currently aiming to create the first type of the CSAR (with `TOSCA.meta`) rather than modifying the specified YAML file (we would need to add `metadata` entry to that file if we wanted to create the second CSAR type). At the moment we create a mixture between the two types as we create `TOSCA.meta` and we also allow only one root level yaml file in the root of the CSAR.

So there are many ways. We can modify current implementation so that we allow multiple root YAML files in the root of the CSAR - here if the user would specify the CSAR entrypoint with `-t/--service-template` it would be ok if there would be multiple YAML files in the root of the CSAR, and if he omits that CLI flag the packager would check if there is only one file in the root to select it as the new entrypoint or if there are multiple it would raise a ParseError like it does now.

Or we can add an additional switch (for instance `--without-meta-file`) for user to decide which CSAR type he would like to receive. What do you propose?

Reply:
Another question: Is `-t/--service-template``okay? Or does `-e/--csar-entrypoint` sound better?

Reply:
As we talked, its currently better to create just the first type of the CSAR with `TOSCA.meta` and keep the `-t/--service-template` option.